content/blog/2018/02/2018-02-22-cheetah.adoc

```diff
@@ -0,0 +1,134 @@
+
+Now I'm _not_ saying this to scare people or to criticize the technology I've helped build. Implementing link:/blog/2017/02/01/pipeline-scalability-best-practice/[Pipeline scalability best practices] coupled with SSD storage keeps Jenkins in a happy place.  We just need context on the weaknesses to see why it's important to address them.
+
+== Introducing Project Cheetah
+
+Today we're announcing the first major results of Project Cheetah, our long-running effort to address these challenges and improve Pipeline scalability.  More broadly, Cheetah aims to help in 3 places:
+
+* *Small-scale containers:* Pipeline needs to run leanly in resource-constrained containers, to enable easy scale-out without consuming excessive resources on shared container hosts.
+* *Enterprise systems:* Pipeline needs to effectively serve high-scale Jenkins instances that are central to many large companies.
+* *General case:* run Pipelines a bit more quickly on average, and allow users to get much-stronger performance in worst-case scenarios.
+
+== Yes, but what does it DO? 
+
+Project Cheetah offers several things, but the most important is Durability Settings for all Pipelines, and especially the Performance-Optimized setting.  This setting avoids several unexpected performance "surprises" that may strike users.  In the general case, it greatly reduces the disk IO needs for Pipeline.  How much?  Below is a graph of storage utilization with legacy Pipeline versions (think early 2017) and with the latest version using the Performance-Optimized mode.  These are tested on an AWS instance backed by an EBS volume provisioned with 300 IOPs. 
+
+Before and After:
+
+image:/images/post-images/2018-02-22-cheetah/io-utilization.png[role="center"]
+
+As you can see, storage utilization goes down by a _lot_.  While the exact number will vary, across the benchmark testcases *this translates to 2x to 6x the Pipeline throughput before becoming IO-bound*. This also increases stability of Jenkins masters because they will tolerate unexpected load.
+
+This comes with a major drop in CPU IOWait as well:
+
+image:/images/post-images/2018-02-22-cheetah/cpu-iowait.png[role="center"]
+
+And of course the rate at which data is written to disk and number of writes/s is also reduced:
+
+image:/images/post-images/2018-02-22-cheetah/diskio.png[role="center"]
+
+For enterprise users, timing stats often show 10-20% of normal builds is serializing the Program and writing the record of steps run ("FlowNodes") - the performance optimized durability setting will cut this to _almost nothing_ (for standard pipelines, 1/100 or less) - so builds will complete faster, especially complex ones.
+
+Please see the link:/doc/book/pipeline/scaling-pipeline/[Pipeline Scalability documentation] for deeper information on the new Durability Settings and how to use them.
```

Comment:
Are you pleading or suggesting. The exact use of the word "please" is a bit foreign to me :)

Reply:
Heh, it's a bit of polite language -- a more gentle way to say "you should go here" or perhaps "don't expect this to answer all questions, you need to look at this other thing too"